sigma_core/models/group.py

```diff
@@ -43,9 +35,84 @@ class Meta:
 
     name = models.CharField(max_length=254)
     visibility = models.CharField(max_length=64, choices=VISIBILITY_CHOICES, default=VIS_PRIVATE)
-    membership_policy = models.CharField(max_length=64, choices=MEMBERSHIP_CHOICES, default=MEMBER_INVITATION)
-    validation_policy = models.CharField(max_length=64, choices=VALIDATION_CHOICES, default=VALID_ADMINS)
     type = models.CharField(max_length=64, choices=TYPE_CHOICES, default=TYPE_BASIC)
 
+    # The permission a member has upon joining
+    # A value of -1 means that no one can join the group.
+    # A value of 0 means that anyone can request to join the group
+    default_member_rank = models.SmallIntegerField(default=-1)
+    # Invite new members on the group
+    req_rank_invite = models.SmallIntegerField(default=1)
+    # Remove a member from the group
+    req_rank_kick = models.SmallIntegerField(default=ADMINISTRATOR_RANK)
+    # Upgrade someone rank 0 to rank 1
+    req_rank_accept_join_requests = models.SmallIntegerField(default=1)
+    # Upgrade other users (up to $yourRank - 1)
+    req_rank_promote = models.SmallIntegerField(default=ADMINISTRATOR_RANK)
+    # Downgrade someone (to rank 1 minimum)
+    req_rank_demote = models.SmallIntegerField(default=ADMINISTRATOR_RANK)
+    # Modify group description
+    req_rank_modify_group_infos = models.SmallIntegerField(default=ADMINISTRATOR_RANK)
+
+    # Related fields:
+    #   - invited_users (model User)
+    #   - memberships (model UserGroup)
+
+    # objects = GroupManager()
+
+    def can_anyone_join(self):
+        return self.default_member_rank >= 0
+
     def __str__(self):
         return "%s (%s)" % (self.name, self.get_type_display())
+
+    # Perms for admin site
+    def has_perm(self, perm, obj=None):
+        return True
+
+    def has_module_perms(self, app_label):
+        return True
+
+    # Permissions
+    @staticmethod
+    def has_read_permission(request):
+        """
+        All groups are visible by default.
+        """
+        return True
+
+    def has_object_read_permission(self, request):
+        """
+        Public groups are visible by everybody. Private groups are only visible by members.
+        """
+        return self.visibility == Group.VIS_PUBLIC or request.user.is_group_member(self)
+
+    @staticmethod
+    def has_write_permission(request):
+        return True
+
+    @staticmethod
+    @allow_staff_or_superuser
+    def has_create_permission(request):
+        """
+        Everybody can create a private. For other types, user must be school admin or sigma admin.
+        """
+        #TODO: Adapt after School model implementation.
+        group_type = request.data.get('type', None)
+        if group_type is not None:
```

Comment:
Why not only
return group_type == Group.TYPE_BASIC
?
If type = None, you can't create the group.